dali/test/python/test_pipeline_debug.py

```diff
@@ -0,0 +1,144 @@
+    if debug:
+        rn50_pipeline_base_debug_values['resized_images'] = resized_images.get()
+    out_type = types.FLOAT16
+
+    output = fn.crop_mirror_normalize(resized_images.gpu(), mirror=rng, device="gpu", dtype=out_type, crop=(
+        224, 224), mean=[0.485 * 255, 0.456 * 255, 0.406 * 255], std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
+    if debug:
+        rn50_pipeline_base_debug_values['output'] = output.get()
+    return rng, jpegs, labels, images, resized_images, output
+
+
+@pipeline_def(batch_size=8, num_threads=3, device_id=0, debug=True)
+def rn50_pipeline():
+    rng = fn.random.coin_flip(probability=0.5, seed=47)
+    print(f'rng: {rng.get().as_array()}')
+    tmp = rng ^ 1
+    print(f'rng xor: {tmp.get().as_array()}')
+    jpegs, labels = fn.readers.file(
+        file_root=file_root, shard_id=0, num_shards=2)
+    if jpegs.get().is_dense_tensor():
+        print(f'jpegs: {jpegs.get().as_array()}')
+    else:
+        print('jpegs shapes:')
+        for j in jpegs.get():
+            print(j.shape())
+    print(f'labels: {labels.get().as_array()}')
+    images = fn.decoders.image(jpegs, device='mixed', output_type=types.RGB)
+    for i in images.get().as_cpu():
+        print(i)
+    for i in images.get():
+        print(i.shape())
+    images = fn.random_resized_crop(images, device="gpu", size=(224, 224), seed=27)
+    for i in images.get():
+        print(i.shape())
+    print(np.array(images.get().as_cpu()[0]))
+    images += 1
+    print(np.array(images.get().as_cpu()[0]))
+    out_type = types.FLOAT16
+
+    output = fn.crop_mirror_normalize(images.gpu(), mirror=rng, device="gpu", dtype=out_type, crop=(
+        224, 224), mean=[0.485 * 255, 0.456 * 255, 0.406 * 255], std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
+    return (output, labels.gpu())
+
+
+@pipeline_def(batch_size=8, num_threads=3, device_id=0)
+def load_images_pipeline():
+    jpegs, labels = fn.readers.file(
+        file_root=file_root, shard_id=0, num_shards=2)
+    images = fn.decoders.image(jpegs, output_type=types.RGB)
+    return images, labels
+
+
+@pipeline_def(batch_size=8, num_threads=3, device_id=0, debug=True)
+def numpy_array_injection_pipeline(images):
+    rng = fn.random.coin_flip(probability=0.5, seed=47)
+    images = fn.random_resized_crop(images, size=(224, 224), seed=27)
+    print(images.get().layout())
+    out_type = types.FLOAT16
+
+    output = fn.crop_mirror_normalize(images.gpu(), mirror=rng, device="gpu", dtype=out_type, crop=(
+        224, 224), mean=[0.485 * 255, 0.456 * 255, 0.406 * 255], std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
+    return (output)
+
+
+def run_pipeline(func, *args, **kwargs):
+    pipe = func(*args, **kwargs)
+    pipe.build()
+    return pipe.run()
+
+
+def test_debug_pipeline_base():
+    rng, jpegs, labels, images, resized_images, output = run_pipeline(rn50_pipeline_base)
+    rng_, jpegs_, labels_, images_, resized_images_, output_ = run_pipeline(
+        rn50_pipeline_base, debug=True)
```

Comment:
There is some test utility called `compare_pipelines` which takes two pipelines, runs them for specified number of iterations and compares the results, you may want to look into that.